### ISSUE TITLE
[VL]  Column name containing parts of Cyrillic cannot be read correctly

### ISSUE THREAD

**Issue by zml1206** [apache/incubator-gluten#6843]
### Backend

VL (Velox)

### Bug description

```
      import testImplicits._
      Seq((1, 2)).toDF("Товары", "овары").write.mode("overwrite").parquet("tmp/t1")
      spark.read.parquet("tmp/t1").show()
      spark.conf.set("spark.gluten.enabled", false)
      spark.read.parquet("tmp/t1").show()
```
enable gluten
```
+------+-----+
|Товары|овары|
+------+-----+
|  null|    2|
+------+-----+
```
disable gluten
```
+------+-----+
|Товары|овары|
+------+-----+
|     1|    2|
+------+-----+
```

### Spark version

None

**Comment by zml1206** [comment]
@zhztheplayer @rui-mo 
Is this a velox problem? Can it be solved by native or disable the scan of column names containing Cyrillic?

**Comment by rui-mo** [comment]
Hi @zml1206, are you saying a full Cyrillic name can be read correctly while a mixed name cannot?

**Comment by zml1206** [comment]
No, it’s just that some Cyrillic letters cannot be parsed, for example "Т"

**Comment by rui-mo** [comment]
Could you please check the written file's content to determine whether the problem is on read or write?

**Comment by zml1206** [comment]
Confirmed problem that it is read.

**Comment by zml1206** [comment]
Roman numerals are not correct either, for example col name `国Ⅵ`.

**Comment by rui-mo** [comment]
The non-ASCII characters are not well supported in Velox tokenizer. Opened https://github.com/facebookincubator/velox/issues/10796 to discuss its support. Thanks.

**Comment by rui-mo** [comment]
There is one remaining issue on the case-sensitivity, seeing https://github.com/facebookincubator/velox/issues/11092. Currently only lower-case non-ASCII field names can be supported in Gluten.
